src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -85,27 +37,27 @@ internal KeyValue(K key, V value)
         [DataMember(IsRequired = true)]
         public K Key
         {
-            get { return _key; }
-            set { _key = value; }
+            get => _key;
+            set => _key = value;
         }
```

Comment:
If you used an auto property, the code would be less confusing.

Reply:
But we need the same field to back multiple properties. So an auto property won't work here. Unless you're suggesting to back a second property with the first. Which doesn't necessarily feel less confusing to me.

Reply:
key = value is what's confusing here. I was suggesting backing the second property with the first, inlining will take care of it all.

Reply:
Moved to auto property. I don't usually like doing so, but I used `this.` while referencing the property for backing the other property - hopefully that jogs the brain into looking at what is going on rather than just glossing over familiar tokens.